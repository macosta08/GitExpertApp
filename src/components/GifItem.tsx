import React from 'react';

interface GifProps {
  id: string;
  title: string;
  url: string;
}

/**
 * `GifItem` es un componente que representa una tarjeta visual para mostrar un GIF con su título.
 *
 * @component
 * @param {GifProps} props - Las propiedades necesarias para renderizar el componente.
 * @param {string} props.id - El identificador único del GIF.
 * @param {string} props.title - El título o descripción del GIF.
 * @param {string} props.url - La URL de la imagen del GIF.
 *
 * @example
 * // Uso básico de GifItem
 * <GifItem id="abc123" title="Sunset in the mountains" url="https://example.com/image.jpg" />
 *
 * @returns {JSX.Element} Un elemento `div` que contiene una imagen y su título.
 */

const GifItem = ({ title, url }: GifProps) => {
  return (
    <div>
      <div className="relative w-full h-0 pb-[100%] overflow-hidden rounded-lg bg-gray-200">
        <img
          src={url}
          alt={title}
          className="absolute inset-0 h-full w-full object-cover object-center group-hover:opacity-75"
        />
      </div>
      <h4 className="mt-4 font-bold text-xl capitalize text-gray-700">
        {title}
      </h4>
    </div>
  );
};

export default GifItem;
