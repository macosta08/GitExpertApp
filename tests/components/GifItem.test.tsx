import { render, screen } from '@testing-library/react';
import GifItem from '../../src/components/GifItem';

describe('Prueba a GifItem', () => {
  const title = 'Pikachu';
  const url =
    'https://media1.giphy.com/media/U2nN0ridM4lXy/giphy.gif?cid=3f1489d2cglbn6k3kljszmp47u58hm1fmr11b5dgdgpigw7a&ep=v1_gifs_search&rid=giphy.gif&ct=g';
  test('debe de hacer match con el snapshot', () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });
  test('debe de mostrar el titulo enviado por props', () => {
    const { getByText } = render(<GifItem title={title} url={url} />);
    expect(getByText(title)).toBeTruthy();
  });
  test('debe mostrar la imagen con el URL y el ALT indicado', () => {
    render(<GifItem title={title} url={url} />);
    screen.debug();
    // expect(screen.getByRole('img').src).toBe(url);
    const { src, alt } = screen.getByRole('img') as HTMLImageElement;
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });
  test('debe mostras titulo', () => {
    render(<GifItem title={title} url={url} />);
    const titleElement = screen.getByText(title) as HTMLElement;
    expect(titleElement).toBeInTheDocument();
    expect(titleElement).toHaveTextContent(title);
  });
  test('debe renderizar los atributos de la imagen correctamente', () => {
    render(<GifItem title={title} url={url} />);
    const imgElement = screen.getByRole('img');
    expect(imgElement).toBeInTheDocument();
    expect(imgElement).toHaveAttribute('src', url);
    expect(imgElement).toHaveAttribute('alt', title);
  });
  test('image tiene el texto alt para la accesibilidad', () => {
    render(<GifItem title={title} url={url} />);
    const imgElement = screen.getByRole('img');
    expect(imgElement).toHaveAttribute('alt', title);
  });
});
