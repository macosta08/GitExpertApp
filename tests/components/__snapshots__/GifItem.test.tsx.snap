// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba a GifItem debe de hacer match con el snapshot 1`] = `
<div>
  <div>
    <div
      class="relative w-full h-0 pb-[100%] overflow-hidden rounded-lg bg-gray-200"
    >
      <img
        alt="Pikachu"
        class="absolute inset-0 h-full w-full object-cover object-center group-hover:opacity-75"
        src="https://media1.giphy.com/media/U2nN0ridM4lXy/giphy.gif?cid=3f1489d2cglbn6k3kljszmp47u58hm1fmr11b5dgdgpigw7a&ep=v1_gifs_search&rid=giphy.gif&ct=g"
      />
    </div>
    <h4
      class="mt-4 font-bold text-xl capitalize text-gray-700"
    >
      Pikachu
    </h4>
  </div>
</div>
`;
